application-licensing-licensor/application-licensing-licensor-api/src/main/java/com/xwiki/licensing/internal/UserCounter.java

```diff
@@ -135,29 +172,23 @@ public long getUserCount() throws Exception
     {
         if (cachedUserCount == null) {
             try {
-                long userCount = 0;
-                for (String wikiId : this.wikiDescriptorManager.getAllIds()) {
-                    userCount += getUserCountOnWiki(wikiId);
-                }
+                Query query = getUserQuery().setLimit(1);
+                long userCount = ((QueryResponse) query.execute().get(0)).getResults().getNumFound();
                 this.logger.debug("User count is [{}].", userCount);
                 this.cachedUserCount = userCount;
-            } catch (WikiManagerException | QueryException e) {
+            } catch (QueryException e) {
                 throw new Exception("Failed to count the users.", e);
             }
         }
 
         return cachedUserCount;
     }
 
-    private long getUserCountOnWiki(String wikiId) throws QueryException
+    private Query getUserQuery() throws QueryException
     {
-        StringBuilder statement = new StringBuilder(", BaseObject as obj, IntegerProperty as prop ");
-        statement.append("where doc.fullName = obj.name and obj.className = 'XWiki.XWikiUsers' and ");
-        statement.append("prop.id.id = obj.id and prop.id.name = 'active' and prop.value = '1'");
-
-        Query query = this.queryManager.createQuery(statement.toString(), Query.HQL);
-        query.addFilter(this.countFilter).setWiki(wikiId);
-        List<Long> results = query.execute();
-        return results.get(0);
+        Query query =
+            queryManager.createQuery("object:XWiki.XWikiUsers AND property.XWiki.XWikiUsers.active:true", "solr");
```

Comment:
It could be interesting to limit the space, e.g., with `space_exact:XWiki` to ensure you don't find user objects in other places.